### ISSUE TITLE
seiso should not raise error if configuration has already been deleted

### ISSUE THREAD

**Issue by adrien-dorsaz** [appuio/seiso#42]
Hello,

Some times, seiso raises this error when running with Gitlab CI and docker image `appuio/oc:v3.11`:

```
$ seiso configmaps -l app="${APPLICATION_NAME}" --older-than "${SEISO_OLDER_THAN}" --keep "${SEISO_KEEP_AT_MOST}" --delete
level=info msg="Seiso 0.7.0, commit a728ac2c7ea214bf9af854d4440d45528c13a5f6, date 2020-09-23T07:25:49Z"
Error: could not delete ConfigMaps for 'xxxx': configmaps "xxx-yyy-9dkg8m2bgh" not found
```

I think seiso could raise a Warning in this case but not an error, as the goal is to delete a config and it seems it's already deleted.

**Comment by ccremer** [comment]
Hi. 

That's interesting, because we are getting the list of candidates directly from the cluster. It seems like at the time where we are querying the cluster and deleting it after filtering, the candidate has been already deleted.... This looks like a race condition.

The fix should be easy, but how exactly do you use Seiso? what is the value of SEISO_OLDER_THAN and SEISO_KEEP_AT_MOST? Do you call seiso immediately after doing a deployment?

**Comment by adrien-dorsaz** [comment]
Depending on which environment is targeted, we've 3 different pair of values.

When I had this error, it was with the "development" environment configured with:

```
SEISO_KEEP_AT_MOST: 2
SEISO_OLDER_THAN: "3d"
```

seiso is run before any new deployment is run, but indeed, we can have some race condition with our integration and/or production environment launch the "seiso + deploy" job.

Maybe we should only run seiso for the development environment ?

Our gitlab-ci was inspired by the one provided by the [appuio's example](https://gitlab.com/appuio/example-django/-/blob/master/.gitlab-ci.yml).

**Comment by ccremer** [comment]
Hm strange, if it runs before the new deployment, I cannot image another reason why there's a race condition without having a closer look into your environment.

In any case, if you are able to wait a bit, the linked PR should ignore not-found-errors (but it will still be printed/logged as "Deleted"). That should remove the need to adapt your pipeline :tm:

**Comment by adrien-dorsaz** [comment]
Cool, many thanks for the PR :-)

**Comment by ccremer** [comment]
We have released 0.7.1 now, please check if that works for you
